src/js/components/Form/Form.js

```diff
@@ -64,14 +64,55 @@ const Form = forwardRef(
       defaultValidationResults,
     );
 
-    useEffect(
-      () => setValidationResults({ errors: errorsProp, infos: infosProp }),
-      [errorsProp, infosProp],
-    );
+    // when onBlur input validation is triggered by a click event,
+    // we need to complete the click event before running the validation
+    // otherwise, click events like reset, etc. may not be registered.
+    // for a detailed scenario/discussion, see: https://github.com/grommet/grommet/issues/4863
```

Comment:
anything causing input to blur, 

add a note: value is the name of the form field waiting to be validated.

Reply:
Done.